core/src/main/java/io/grpc/ServerProvider.java

```diff
@@ -46,6 +46,27 @@ static final ServerProvider load(ClassLoader cl) {
         best = current;
       }
     }
+    
+    if (best == null) {
+      best = loadOsgi();
```

Comment:
Instead, could you do the same `iterator().hasNext()` trick as elsewhere?

```java
ServiceLoader<ServerProvider> providers = ServiceLoader.load(ServerProvider.class, cl);
if (!providers.iterator().hasNext() && classLoader == NameResolverProvider.class.getClassLoader()) {
  providers = ServiceLoader.load(ServerProvider.class);
}
```

Reply:
I think that should be possible, the code there is just structured differently so that would require a bit more refactoring.

Reply:
Ok will do this